src/mpyl/steps/deploy/k8s/chart.py

```diff
@@ -538,6 +543,36 @@ def _get_env_vars(self):
             filter(lambda v: v.get_value(self.target) is not None, self.sealed_secrets)
         )
         sealed_secrets = self._create_sealed_secret_env_vars(sealed_secrets_for_target)
+
+        if (
+            self.step_input.run_properties.target == Target.PULL_REQUEST
+        ):  # and not step_input.dry_run:
+            ranger_config = cluster_config(
+                self.step_input.run_properties.target, self.step_input.run_properties
+            )
+            k8s_config.load_kube_config(context=ranger_config.context)
+            #  Use context="vdb-core-digital-k8s-test-fqdn" for local testing
+            kubernetes_api = CoreV1Api()
+            cluster_secrets: V1SecretList = kubernetes_api.list_namespaced_secret(
+                namespace=self.deployment.namespace  # Should be source namespace?
+            )
+            secrets_to_copy: list[V1Secret] = list(
+                (
+                    cluster_secret
+                    for cluster_secret, defined_secret in zip(
+                        cluster_secrets.items, self.secrets
+                    )
+                    if cluster_secret.metadata.name
+                    == defined_secret.value_from.get("secretKeyRef").get("name")
+                )
+            )
+            pr_namespace = "cypress"  # Should be pr namespace
```

Comment:
I started checking locally but I think it's a bit too much of a hassle. I think we should just create a prototype generic implementation and test with a real PR in a real test deployment